src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1726,7 +1749,10 @@ class AliasExpressionSegment(BaseSegment):
             ),
             # just a table alias
             Ref("SingleIdentifierGrammar"),
-            exclude=Ref("JoinTypeKeywords"),
+            exclude=OneOf(
+                Ref.keyword("LATERAL"),
```

Comment:
You may not need the `Ref.keyword` now. You do need it, if it's the ONLY item in the `exclude` paramer as it needs a match, but think it will match agains the `OneOf` now without the need of that.

```suggestion
                "LATERAL",
```